proxy/src/control/destination/background.rs

```diff
@@ -33,24 +33,53 @@ use control::{
     cache::{Cache, CacheChange, Exists},
     fully_qualified_authority::FullyQualifiedAuthority,
     remote_stream::{Receiver, Remote},
-    AddOrigin, Backoff, LogErrors
+        AddOrigin, Backoff, LogErrors
 };
 use dns::{self, IpAddrListFuture};
 use telemetry::metrics::DstLabels;
 use transport::{DnsNameAndPort, HostAndPort, LookupAddressAndConnect};
 use timeout::Timeout;
 use transport::tls;
 
-type DestinationServiceQuery<T> = Remote<PbUpdate, T>;
-type UpdateRx<T> = Receiver<PbUpdate, T>;
+pub trait GetDst {
+    type Error: fmt::Debug;
+    type Future: Future<
+        Item = grpc::Response<Self::Stream>,
+        Error = grpc::Error<Self::Error>,
+    >;
+    type Stream: Stream<Item = PbUpdate, Error = grpc::Error>;
+
+    fn poll_ready(&mut self) -> Poll<(), grpc::Error<Self::Error>>;
+
+    fn get_dst(&mut self, dst: Destination) -> Self::Future;
```

Comment:
Nit: I'm not a big fan of the name of this function (or the trait, TBQH). I'd expect something named `get_dst` to _return_ a `Destination`... maybe `lookup` or `lookup_dst` would be clearer?

Reply:
OTTOMH I'd probably call the function either `lookup` or `query; or `lookup_dst`/`query_dst` if you think that's necessary (I think the type signature of the function makes it fairly clear that it takes in a destination). 

For the trait, I suppose `LookupDst` or `QueryDst` would be fine...trying to come up with something more descriptive and all I can think of off the top of my head is `DestinationProvider`, which is...not great.

Anyway, @olix0r, don't feel obligated to use any of my suggestions if you can think of any more descriptive names. I'm just spitballing here.